core/cmd/app.go

```diff
@@ -187,16 +196,55 @@ func NewApp(client *Client) *cli.App {
 				},
 			},
 			Before: func(c *cli.Context) error {
-				if client.flagsProcessed {
-					if c.IsSet("config") || c.IsSet("secrets") {
-						// invalid mix of flags here and root
-						return fmt.Errorf("multiple commands with --config or --secrets flags. only one command may specify these flags. when secrets are used, they must be specific together in the same command")
+				errNoDuplicateFlags := fmt.Errorf("multiple commands with --config or --secrets flags. only one command may specify these flags. when secrets are used, they must be specific together in the same command")
+				if c.IsSet("config") {
+					if client.configFilesIsSet || client.secretsFileIsSet {
+						return errNoDuplicateFlags
+					} else {
+						client.configFiles = c.StringSlice("config")
+					}
+				}
+
+				if c.IsSet("secrets") {
+					if client.configFilesIsSet || client.secretsFileIsSet {
+						return errNoDuplicateFlags
+					} else {
+						client.secretsFile = c.String("secrets")
 					}
-					// flags at root
-					return nil
 				}
+
 				// flags here, or ENV VAR only
-				return client.initConfigAndLogger(&opts, c.StringSlice("config"), c.String("secrets"))
+				err := client.initServerConfig(&opts, client.configFiles, client.secretsFile)
+				if err != nil {
+					return err
+				}
+
+				logFileMaxSizeMB := client.Config.LogFileMaxSize() / utils.MB
+				if logFileMaxSizeMB > 0 {
```

Comment:
what is the meaning and intention if it's 0? i would have thought =0 is either an error or it means unlimited and we would still need to ensure the permissions